graph/appender/unused_service.go

```diff
@@ -47,25 +48,30 @@ func addUnusedNodes(trees *[]*tree.ServiceNode, namespaceName string, deployment
 	}
 }
 
-func buildStaticNodeList(namespaceName string, deployments *v1beta1.DeploymentList) []*tree.ServiceNode {
+func buildStaticNodeList(namespaceName string, pods *v1.PodList) []*tree.ServiceNode {
 	nonTrafficList := make([]*tree.ServiceNode, 0)
+	resolvedServices := make(map[string]bool)
 	appLabel := config.Get().ServiceFilterLabelName
 	versionLabel := config.Get().VersionFilterLabelName
 	identityDomain := config.Get().Products.Istio.IstioIdentityDomain
-	for _, deployment := range deployments.Items {
-		app, ok := deployment.GetObjectMeta().GetLabels()[appLabel]
+	for _, pod := range pods.Items {
+		app, ok := pod.GetObjectMeta().GetLabels()[appLabel]
 		if !ok {
-			log.Warningf("Deployment %s has not a proper app label [%s]", deployment.Name, appLabel)
+			log.Warningf("Pod %s has not a proper app label [%s]", pod.Name, appLabel)
 			continue
 		}
-		version, ok := deployment.GetObjectMeta().GetLabels()[versionLabel]
+		version, ok := pod.GetObjectMeta().GetLabels()[versionLabel]
 		if !ok {
-			log.Warningf("Deployment %s has not a proper version label [%s]", deployment.Name, versionLabel)
+			log.Warningf("Pod %s has not a proper version label [%s]", pod.Name, versionLabel)
```

Comment:
Are we sure we should expect app and version labels on all pods in the namespaces?  If we can filter in advance, as mentioned above, then we can assume the labels exist.  And if we can't filter I'm not sure we should log warnings, but rather just discard the pod.

Reply:
Now that the retrieved pods are guaranteed to have the app and version labels, I removed these checks.